### DISCUSSION TITLE
Slickgrid Gantt

### DISCUSSION THREAD

**Discussion by zewa666** [ghiscoding/slickgrid-universal#1723] [Ideas]
Hey there, long time no see @ghiscoding. Been quite busy with other stuff, but in a couple of weeks my next Grid sprints are starting (yeeey).

I wanted to share a thing that we'll perhaps pursue for a new project, where we want to use a Gantt chart. But instead of those default minimalistic side-grids, we'd like to leverage the full power of slickgrid.

Here's my very quick and rough prototype

![frappe-gantt](https://github.com/user-attachments/assets/9e62ac66-29b1-463c-b710-74c29e2906d7)

So as you can see I'm using https://frappe.io/gantt and managed to just quickly sync the left-side scrolling while keeping scrolling within the Gantt control unbound. In combo with our OData Backend I'd also sync the values for both the grid/chart as well as forward changes from the grid or chart vice versa. That means there's more work to do but I'm already quite confident that this might work out quite nicely.

**Comment by ghiscoding** [comment]
oh wow that is quite nice to see in action, so you're just synching with the `onScroll`, I would surely welcome this new example :)
You talked about the OData and Pagination so I figured that you can simply subscribe to the `onPaginationChanged` event and then I went on the [Available Events](https://ghiscoding.gitbook.io/slickgrid-universal/events/available-events#pagination-service) docs page and realized all these events weren't being listed for that service, so I just added them, basically here they are:

#### Pagination Service
  - `onBeforePaginationChange`
  - `onPaginationChanged`
  - `onPaginationRefreshed`
  - `onPaginationPresetsInitialized`
  - `onPaginationVisibilityChanged`
  - `onPaginationSetCursorBased` (for GraphQL only)

It's quite cool to see all these ideas and concepts that you guys come up with, it's awesome to know real world usage with such useful UX are possible with SlickGrid :)

I bet they will ask you to customize the Excel Export as well for that lol, and who knows, it might be possible with what we expose from excel-builder-vanilla 😆

**Reply by zewa666** [reply]
pssst, we dont talk about excel over here ;) trying to get the customers off excel and over to slickgrid :)

yep its merely syncing the onScroll now and making sure the chart has the same row height so that a scrollTop sync is enough. 

with odata its even simpler as the backendService' s complete handler is the ideal place to perform a sync for all things odata loading wise.

and ofc once all is set, I'll create a PR for a new example.

**Comment by ghiscoding** [comment]
Talking about spreading SlickGrid love... it looks like Microsoft is now using Slickgrid-Universal (via Slickgrid-React) in 1 of their VSCode extension [vscode-cosmosdb](https://github.com/microsoft/vscode-cosmosdb), well that's a great achievement :)

See 1 of their [PR](https://github.com/microsoft/vscode-cosmosdb/pull/2327)

Pop the Champagne I guess 🍾 😆 
![image](https://github.com/user-attachments/assets/38efbe61-59ff-46a4-9ca6-09d3d5607014)

**Reply by zewa666** [reply]
noice styling, somewhat fits the FluentUi patterns.
